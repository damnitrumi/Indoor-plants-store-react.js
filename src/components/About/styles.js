import styled, { css } from "styled-components";
import { Container as SectionWrapper } from "../SectionWrapper/styles";
import { Container as Text } from "../Text/styles";

export const Container = styled.div`
  ${() => css`
    counter-reset: TextCounter;
    ${SectionWrapper} {
      display: flex;
      flex-direction: column;
      gap: 180px;
    }
  `}
`;
export const AboutContainer = styled.div`
  ${({ theme }) => css`
    ${Text}::before {
      content: counter(TextCounter) ".";
      counter-increment: TextCounter;
      font-size: ${theme.font.sizes.large};
      font-family: ${theme.font.family.primary};
      margin-right: ${theme.spacings.small};
      color: ${theme.colors.lightGreen};
    }

    padding: ${theme.spacings.small} 0;
    /* margin-top: 180px; */
    overflow-y: auto;
  `}
`;
