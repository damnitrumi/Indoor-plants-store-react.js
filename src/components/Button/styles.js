import styled, { css } from "styled-components";

export const Container = styled.button`
  ${({ theme }) => css`
    background: transparent;
    border: none;
    cursor: pointer;
    padding: ${theme.spacings.tiny};
    color: white;
    position: relative;
  `}
`;
