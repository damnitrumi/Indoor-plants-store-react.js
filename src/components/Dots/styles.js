import styled, { css } from "styled-components";

export const Container = styled.span`
  ${({ theme, current, index }) => css`
    display: inline-block;
    border-radius: 50%;
    height: ${theme.spacings.tiny};
    width: ${theme.spacings.tiny};
    background-color: ${current == index
      ? theme.colors.grey
      : theme.colors.mediumGrey};
  `}
`;
