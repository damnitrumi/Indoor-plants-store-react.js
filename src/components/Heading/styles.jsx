import styled, { css } from "styled-components";

export const Container = styled.h1`
  ${({ theme }) => css`
    font-family: ${theme.font.family.primary};
    color: ${theme.colors.darkGreen};
    margin-bottom: ${theme.spacings.medium};
    font-size: ${theme.font.sizes.large};
    text-align: center;
  `}
`;
