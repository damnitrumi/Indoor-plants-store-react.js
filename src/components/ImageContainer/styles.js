import styled, { css } from "styled-components";

export const Container = styled.img`
  ${() => css``}
`;
