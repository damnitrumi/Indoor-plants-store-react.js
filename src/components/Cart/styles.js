import styled, { css } from "styled-components";
import { Container as SectionWrapper } from "../SectionWrapper/styles";
import { Container as Button } from "../Button/styles";
import { Container as Heading } from "../Heading/styles";

export const Container = styled.div`
  ${({ theme, visible }) => css`
    height: 100vh;
    transition: transform 0.5s;
    transform: translateX(100%);
    position: absolute;
    top: 0;
    width: 360px;
    ${visible &&
    css`
      transform: translateX(0);
    `}
    ${SectionWrapper} {
      display: flex;
      flex-direction: column;
      > ${Button} {
        text-align: center;
        background-color: ${theme.colors.lightGreen};
        display: block;
        width: 85%;
        margin: 0 auto;
        border-radius: 5px;
        font-size: ${theme.font.sizes.small};
      }

      > ${Heading} {
        font-size: ${theme.font.sizes.medium};
      }
    }
  `}
`;

export const ItensContainer = styled.div`
  ${() => css`
    overflow-y: auto;
    flex: 1;
  `}
`;
