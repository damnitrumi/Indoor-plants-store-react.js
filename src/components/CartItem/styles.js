import styled, { css } from "styled-components";

import { Container as Heading } from "../Heading/styles";
import { Container as Button } from "../Button/styles";

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    margin-top: ${theme.spacings.small};
    ${Heading} {
      margin-bottom: 0;
      font-size: 1.4rem;
      font-family: ${theme.font.family.secondary};
    }
    ${Button} {
      color: initial;
    }
  `}
`;

export const ProductImageContainer = styled.div`
  ${() => css`
    flex: 1;
  `}
`;
export const ProductData = styled.div`
  ${({ theme }) => css`
    flex: 2;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    padding-right: ${theme.spacings.large};
  `}
`;

export const NameContainer = styled.div`
  ${() => css`
    display: flex;
    justify-content: space-between;
  `}
`;

export const ProductOptionsContainer = styled.div`
  ${() => css``}
`;

export const QuantityContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: space-between;
    font-size: ${theme.font.sizes.medium};
  `}
`;
export const Quantity = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    border: 1px solid rgba(0, 0, 0, 0.3);
    ${Button}:first-of-type {
      border-right: 1px solid rgba(0, 0, 0, 0.3);
    }

    ${Button}:last-of-type {
      border-left: 1px solid rgba(0, 0, 0, 0.3);
    }

    span {
      padding: 0 ${theme.spacings.tiny};
    }
  `}
`;
