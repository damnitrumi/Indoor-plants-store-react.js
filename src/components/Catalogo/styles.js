import styled, { css } from "styled-components";
import { Container as SectionWrapper } from "../SectionWrapper/styles";
import { Container as ImageContainer } from "../ImageContainer/styles";

// export const SectionContainer = styled.div`
//   ${() => css`
//     display: flex;
//     flex-direction: column;
//     height: 100%;
//   `}
// `;

export const Container = styled.div`
  ${({ visible }) => css`
    height: 100vh;
    display: flex;
    flex-direction: column;
    transition: transform 0.5s;
    transform: translateX(-100%);
    position: absolute;
    width: 360px;
    ${visible &&
    css`
      transform: translateX(0);
    `}
    ${SectionWrapper} {
      flex: 1;
      display: flex;
      flex-direction: column;
    }
  `}
`;

export const SlidesContainer = styled.div`
  ${() => css`
    /* ${ImageContainer} {
      height: 460px;
    } */
  `}
`;

export const SliderContainer = styled.div`
  ${() => css`
    flex: 1;
  `}
`;

export const BgContainer = styled.span`
  ${({ theme }) => css`
    display: block;
    background-color: ${theme.colors.lightGrey};
    width: 230px;
    height: 300px;
    border-radius: ${theme.spacings.medium};
    position: absolute;
    z-index: -1;
    top: 40px;
  `}
`;

export const DotsContainer = styled.div`
  ${() => css`
    display: flex;
    justify-content: center;
    gap: 3px;
    position: absolute;
    left: 150px;
    bottom: 290px;
  `}
`;
