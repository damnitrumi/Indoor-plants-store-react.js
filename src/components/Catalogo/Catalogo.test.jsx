import { screen } from "@testing-library/react";
import { renderTheme } from "../../styles/render-theme";
import { Catalogo } from ".";

import plantsMock from "../utils/plantsMock";

describe("<Catalogo />", () => {
  it("should render Catalogo component", () => {
    renderTheme(<Catalogo plants={plantsMock} />);

    // Aqui eu sei que ele deveria renderizar as imagens na tela na quantidade de elementos no array recebido + as 3 imagens do Menu, vou usar isso para testar se está renderizando corretamente

    expect(screen.getAllByRole("img")).toHaveLength(plantsMock.length + 3);
  });

  it("should render Catalogo component without data", () => {
    renderTheme(<Catalogo />);

    // Aqui eu sei que ele deveria renderizar as imagens na tela na quantidade de elementos no array recebido + as 3 imagens do Menu, vou usar isso para testar se está renderizando corretamente
    // E no caso nesse em específico, não estou passando nenhum elemento, logo só deve renderizar as 3 imagens do Menu
    expect(screen.getAllByRole("img")).toHaveLength(3);
  });

  it("should render", () => {
    const { container } = renderTheme(<Catalogo plants={plantsMock} />);

    expect(container).toMatchSnapshot();
  });
});
