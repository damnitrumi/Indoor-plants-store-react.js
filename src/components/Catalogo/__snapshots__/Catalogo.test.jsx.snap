// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Catalogo /> should render 1`] = `
.c2 {
  height: 100vh;
  border-radius: 5px;
  background: white;
  padding: 3.2rem;
}

.c0 {
  position: relative;
  z-index: 1;
  height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transition: -webkit-transform 0.5s;
  -webkit-transition: transform 0.5s;
  transition: transform 0.5s;
  -webkit-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
  position: absolute;
  top: 0;
  width: 360px;
}

.c0 .c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c10 {
  display: block;
  background-color: #f6f4f4;
  width: 230px;
  height: 300px;
  border-radius: 2.4rem;
  position: absolute;
  z-index: -1;
  top: 40px;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 3px;
  position: absolute;
  left: 150px;
  bottom: 290px;
}

.c9 {
  -webkit-transition: -webkit-transform 0.2s ease-out;
  -webkit-transition: transform 0.2s ease-out;
  transition: transform 0.2s ease-out;
}

.c9 div {
  padding: 1rem;
  height: 100%;
  width: 0px;
  height: 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c9 img {
  max-width: 100%;
  max-height: 100%;
}

.c8 {
  all: initial;
  width: 100%;
  height: 100%;
  max-height: 100vh;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  will-change: transform,scale;
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
}

.c8 .slide-outer {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  overflow: hidden;
  width: 100%;
  height: 100%;
  max-height: 100vh;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  background: transparent;
  border: none;
  cursor: pointer;
  padding: 0.8rem;
  color: white;
  position: relative;
}

.c17 {
  font-family: 'Aclonica',sans-serif;
  color: #213e3a;
  margin-bottom: 2.4rem;
  font-size: 3.2rem;
  text-align: center;
}

.c24 {
  margin-bottom: 2.4rem;
  font-size: 1.6rem;
  font-family: 'Montserrat',sans-serif;
  text-align: justify;
}

.c14 {
  background-color: #303030;
  width: 100%;
  -webkit-flex: 0.7;
  -ms-flex: 0.7;
  flex: 0.7;
  border-radius: 3.2rem 3.2rem 0 0;
  padding: 3.2rem 4.0rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c14 .c16 {
  color: white;
  margin-bottom: 0;
  font-size: 1.6rem;
}

.c14 > .c4 {
  text-align: center;
  background-color: #009d84;
  display: block;
  width: 85%;
  margin: 0 auto;
  border-radius: 5px;
  font-size: 1.6rem;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 2.4rem;
}

.c18 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 0.8rem;
  position: relative;
}

.c22 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c22 .c23 {
  padding: 0.8rem;
  color: #c7c8c8;
  font-size: 1.3rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  max-height: 60px;
  overflow: auto;
}

.c20 {
  position: relative;
  -webkit-letter-spacing: 0.05rem;
  -moz-letter-spacing: 0.05rem;
  -ms-letter-spacing: 0.05rem;
  letter-spacing: 0.05rem;
}

.c20::after {
  display: none;
  content: "";
  background-color: #1b6359;
  width: 100%;
  height: 3px;
  position: absolute;
  left: -1%;
  display: block;
}

.c21 {
  position: relative;
  -webkit-letter-spacing: 0.05rem;
  -moz-letter-spacing: 0.05rem;
  -ms-letter-spacing: 0.05rem;
  letter-spacing: 0.05rem;
}

.c21::after {
  display: none;
  content: "";
  background-color: #1b6359;
  width: 100%;
  height: 3px;
  position: absolute;
  left: -1%;
}

.c12 {
  display: inline-block;
  border-radius: 50%;
  height: 0.8rem;
  width: 0.8rem;
  background-color: #303030;
}

.c13 {
  display: inline-block;
  border-radius: 50%;
  height: 0.8rem;
  width: 0.8rem;
  background-color: #c7c8c8;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 c2"
    >
      <nav
        class="c3"
      >
        <button
          class="c4 c5"
        >
          <img
            class=""
            src="/assets/images/menu-dark.png"
          />
        </button>
        <button
          class="c4 c5"
        >
          <img
            class=""
            src="/assets/images/short-logo-dark.png"
          />
        </button>
        <button
          class="c4 c5"
        >
          <img
            class=""
            src="/assets/images/cart-dark.png"
          />
        </button>
      </nav>
      <div
        class="c6"
      >
        <div
          class="c7 SliderWrapper"
        >
          <div
            class="c8 SliderStyles"
            style="transform: translateX(0px);"
          >
            <div
              class="slide-outer"
            >
              <div
                class="c9 SlideStyles"
              >
                <div
                  class="slide-inner"
                  unselectable="on"
                >
                  <div
                    class=""
                  >
                    <img
                      class=""
                      src="/assets/images/plant.png"
                    />
                    <span
                      class="c10"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="slide-outer"
            >
              <div
                class="c9 SlideStyles"
              >
                <div
                  class="slide-inner"
                  unselectable="on"
                >
                  <div
                    class=""
                  >
                    <img
                      class=""
                      src="/assets/images/plant.png"
                    />
                    <span
                      class="c10"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="slide-outer"
            >
              <div
                class="c9 SlideStyles"
              >
                <div
                  class="slide-inner"
                  unselectable="on"
                >
                  <div
                    class=""
                  >
                    <img
                      class=""
                      src="/assets/images/plant.png"
                    />
                    <span
                      class="c10"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="slide-outer"
            >
              <div
                class="c9 SlideStyles"
              >
                <div
                  class="slide-inner"
                  unselectable="on"
                >
                  <div
                    class=""
                  >
                    <img
                      class=""
                      src="/assets/images/plant.png"
                    />
                    <span
                      class="c10"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="slide-outer"
            >
              <div
                class="c9 SlideStyles"
              >
                <div
                  class="slide-inner"
                  unselectable="on"
                >
                  <div
                    class=""
                  >
                    <img
                      class=""
                      src="/assets/images/plant.png"
                    />
                    <span
                      class="c10"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c11"
    >
      <span
        class="c12"
      />
      <span
        class="c13"
      />
      <span
        class="c13"
      />
      <span
        class="c13"
      />
      <span
        class="c13"
      />
    </div>
    <div
      class="c14"
    >
      <div
        class="c15"
      >
        <h1
          class="c16 c17"
        >
          Cacto Plant 1
        </h1>
        <h1
          class="c16 c17"
        >
          R$:21.00
        </h1>
      </div>
      <div
        class="c18"
      >
        <div
          class="c19"
        >
          <button
            class="c4 c5 c20"
          >
            Details
          </button>
          <button
            class="c4 c5 c21"
          >
            Reviews
          </button>
          <button
            class="c4 c5 c21"
          >
            About
          </button>
        </div>
        <div
          class="c22"
        >
          <p
            class="c23 c24"
          >
            Cactus are succulent perennial plants 1
          </p>
        </div>
      </div>
      <button
        class="c4 c5"
      >
        Add To Cart
      </button>
    </div>
  </div>
</div>
`;
